mm2src/coins/solana/solana_tests.rs

```diff
@@ -335,3 +337,112 @@ fn solana_test_tx_history() {
+            maker_secret_hash: secret_hash.as_slice(),
+        },
+        swap_contract_address: &Some(Bytes::from(solana_program_id)),
+        swap_unique_data: pk_data.as_slice(),
+        watcher_reward: false,
+    };
+    let refund_tx = block_on(coin.send_maker_refunds_payment(refund_args)).unwrap();
+    println!("refund tx {:?}", refund_tx);
+}
+
+#[test]
+fn solana_coin_send_and_spend_maker_payment() {
+    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+    let (_, coin) = solana_coin_for_test(passphrase, SolanaNet::Devnet);
+    let solana_program_id = "AEwou7VU9bahWRSN2zzgMGqozNY2G27U9fbbdLF9myZx";
+    let solana_program_id = bs58::decode(solana_program_id).into_vec().unwrap_or_else(|e| {
+        eprintln!("Failed to decode program ID: {}", e);
+        Vec::new()
+    });
+
+    let pk_data = [1; 32];
+    let lock_time = now_sec() - 1000;
+    let taker_pub = coin.key_pair.pubkey().to_string();
+    let taker_pub = Pubkey::from_str(taker_pub.as_str()).unwrap();
+    let secret = [0; 32];
+    let secret_hash = sha256(&secret);
+
+    let maker_payment_args = SendPaymentArgs {
+        time_lock_duration: 0,
+        time_lock: lock_time,
+        other_pubkey: taker_pub.as_ref(),
+        secret_hash: secret_hash.as_slice(),
+        amount: "0.01".parse().unwrap(),
+        swap_contract_address: &Some(Bytes::from(solana_program_id.clone())),
+        swap_unique_data: &[],
+        payment_instructions: &None,
+        watcher_reward: None,
+        wait_for_confirmation_until: 0,
+    };
+
+    let tx = coin.send_maker_payment(maker_payment_args).wait().unwrap();
+    println!("swap tx {:?}", tx);
+
+    let maker_pub = taker_pub;
+
+    let spends_payment_args = SpendPaymentArgs {
+        other_payment_tx: &tx.tx_hex(),
+        time_lock: lock_time,
+        other_pubkey: maker_pub.as_ref(),
+        secret: &secret,
+        secret_hash: secret_hash.as_slice(),
+        swap_contract_address: &Some(Bytes::from(solana_program_id)),
+        swap_unique_data: pk_data.as_slice(),
+        watcher_reward: false,
+    };
+
+    let spend_tx = block_on(coin.send_taker_spends_maker_payment(spends_payment_args)).unwrap();
+    log!("spend tx {}", hex::encode(spend_tx.tx_hash_as_bytes().0));
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_get_public_key() {
+    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
+    let (_, coin) = solana_coin_for_test(passphrase, SolanaNet::Devnet);
+    let expected_pubkey = coin.key_pair.pubkey().to_string();
+    let result = coin.get_public_key().await;
+    match result {
+        Ok(pubkey) => assert_eq!(pubkey, expected_pubkey),
+        Err(e) => panic!("Expected Ok, got Err: {:?}", e),
+    }
+}
```

Comment:
This test can be better written as below, but I don't think it's actually needed, please just remove it.
```suggestion
#[test]
fn test_get_public_key() {
    let passphrase = "federal stay trigger hour exist success game vapor become comfort action phone bright ill target wild nasty crumble dune close rare fabric hen iron".to_string();
    let (_, coin) = solana_coin_for_test(passphrase, SolanaNet::Devnet);
    let expected_pubkey = coin.key_pair.pubkey().to_string();
    let pubkey = match block_on(coin.get_public_key()) {
        Ok(value) => value,
        Err(e) => panic!("Failed to get public key: {:?}", e),
    };
    assert_eq!(pubkey, expected_pubkey);
}
```